src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/RequestExecutionQueue.cs

```diff
@@ -297,6 +281,12 @@ private async Task ProcessQueueAsync()
         }
     }
 
+    /// <summary>
+    /// Choose the method handler for the given request. By default this uses the <see cref="IHandlerProvider"/>.
+    /// </summary>
+    protected virtual IMethodHandler GetHandlerForRequest(IQueueItem<TRequestContext> work)
+        => _handlerProvider.GetMethodHandler(work.MethodName, work.RequestType, work.ResponseType);
```

Comment:
this seems like a bug.  it's not passing any sort of language along.  (which def makes me think this has to be non-optional).